pkg/visor/init.go

```diff
@@ -487,28 +488,66 @@ func initUptimeTracker(v *Visor) bool {
 	return true
 }
 
-func initTrustedVisors(v *Visor) bool {
-	const trustedVisorsTransportType = tptypes.STCPR
+func initPublicVisors(v *Visor) bool {
+	if !v.conf.Transport.AutoconnectPublic {
+		return true
+	}
+	log := v.MasterLogger().PackageLogger("public_visors")
+	proxyDisc := v.conf.Launcher.Discovery.ServiceDisc
+	if proxyDisc == "" {
+		proxyDisc = skyenv.DefaultServiceDiscAddr
+	}
 
-	go func() {
-		time.Sleep(transport.TrustedVisorsDelay)
-		for _, pk := range v.tpM.Conf.DefaultVisors {
-			v.log.WithField("pk", pk).Infof("Adding trusted visor")
-
-			if _, err := v.tpM.SaveTransport(context.Background(), pk, trustedVisorsTransportType); err != nil {
-				v.log.
-					WithError(err).
-					WithField("pk", pk).
-					WithField("type", trustedVisorsTransportType).
-					Warnf("Failed to add transport to trusted visor via")
-			} else {
-				v.log.
-					WithField("pk", pk).
-					WithField("type", trustedVisorsTransportType).
-					Infof("Added transport to trusted visor")
-			}
+	_, portStr, err := net.SplitHostPort(v.conf.STCP.LocalAddr)
```

Comment:
Why is the local address of a visor needed in this case. Setting up `stcpr` should work without the IP address of the local or the remote being specified in config. Usually no one defines teh `stcp` config fields because its manual and error prone wherefore we added `stcpr` to avoid having to define that manually.

Reply:
It's been like that in commits before me, I only tested that it works and restructured it later.
I read through the code a bit and I believe the problem is that appdisc client needs port value to be constructed, which in turn is required because every such client also advertises its service. It currently doesn't support just fetching public services. Needs closer look